pepdbagent/modules/project.py

```diff
@@ -558,13 +605,27 @@ def update(
                             f"pephub_sample_id '{PEPHUB_SAMPLE_ID_KEY}' is missing in samples."
                             f"Please provide it to update samples, or use overwrite method."
                         )
+                    if len(update_dict["samples"]) > MAX_HISTORY_SAMPLES_NUMBER:
+                        _LOGGER.warning(
+                            f"Number of samples in the project exceeds the limit of {MAX_HISTORY_SAMPLES_NUMBER}."
+                            f"Samples won't be updated."
+                        )
+                        new_history = None
```

Comment:
If it's not too complicated, can we just delete the oldest one and then insert?

Reply:
Also, does this warrant an `Exception`? I guess a warning is fine, but if the user is not paying attention they might not realize their samples histories aren't being updated 😅